### ISSUE TITLE
Rework of the error handling.

### ISSUE THREAD

**Issue by TheTmirror** [tu-dortmund-ls12-rt/rtems#12]
### Rework of the error handling.
When working with the current error handling and extending it from time to time it is noticed, that the handling is in sometimes not consistent. When more important tasks are done, the error handling should be worked on. There needs to be a clear definition when, for example, an error is argument dependent or not, and when it needs to be returned cascadingly.

### Current recommendation without in depth thinking.
In general it is suggested, that the error handling is always done soley from within the viewpoint of the respective function. Furthermore it should be categorized if an error is either directly dependent on the arguments of the respective function, indirectly dependent (for example a NULL POINTER dependent on a casted argument) or not dependent at all (then it would be an internal error).
The bottom line is, that it needs to be consistent.

**Comment by TheTmirror** [comment]
### Merged issue into this - Update 18.03.2022
The issue #11 was merged into this issue.

**Comment by TheTmirror** [comment]
### Finished this issue - Update 13.05.2022
This issue was finished in commit 6f90c79db17aab30fa4f39fab086705e05e7e14e. Afterwards it was fully merged into branch DEV.
